base/reflection.jl

```diff
@@ -1050,32 +1051,70 @@ function MethodList(mt::Core.MethodTable)
     return MethodList(ms, mt)
 end
 
+# we are not actually returning the method table, but an array of methods
+# besides adding the documentation related to `:callable` I'm also rewording
+# Return the method table for `f`. ----> Return an array of methods for `f`.
 """
-    methods(f, [types], [module])
+    methods(f, [types], [module], [:callable])
 
-Return the method table for `f`.
+Return an array of methods for `f`.
 
 If `types` is specified, return an array of methods whose types match.
 If `module` is specified, return an array of methods defined in that module.
 A list of modules can also be specified as an array.
 
+If `:callable` key is used, return the array of methods defined for the callable object of type `f`.
```

Comment:
This is not exactly what is implemented. Currently, `methods(f, :callable)` is also allowed if `f` is not a type, and in that case gives the same result as `methods(f)`. What about dropping this option? Then `:callable` would be defined by
```
methods(f) == methods(typeof(f), : callable)
```
for any argument `f`.
```
By the way, would it simplify the implementation to make `callable` (or whatever the name should be) a keyword argument?

Reply:
> This is not exactly what is implemented. Currently, methods(f, :callable) is also allowed if f is not a type, and in that case gives the same result as methods(f). What about dropping this option?

I thought that using `:callable` together with an instance would have the same behavior as `methods(instance)`. 

This is why `methods(f) == methods(f, :callable)` at this point (when `f` is either a function or an instance).

But even more, `methods(f) == methods(typeof(f), : callable)` holds even if `f` is a function (maybe it would be a good idea to extend the tests a little more:
```julia
f(x::Int) = x + 1

methods(f) == methods(f, :callable) # true
methods(f) == methods(typeof(f), :callable) # true
```
Functions are treated as instances given the fact that the following are equivalent:
```julia
(::typeof(f))(x::Float64) = x + 10.0
# above the same as below
f(x::Float64) = x + 10.0
```
Implementation wise is easier to just throw if `:callable` is used on an instance - however, my initial thought was to make it *forgiving* by offering the following equality: `methods(f) == methods(f, :callable)`. 

However, I want to make clear that I am open to suggestions and willing to make changes as long as we can pass the main feature in the `Base`. Because right now is pretty complicated (and not supported by `methods`) to retrieve the methods for callable objects without actually using an instance.